test/test_compile_cache.py

```diff
@@ -215,6 +215,88 @@ def _nop_compile(x, _):
             args = [torch.randn(10, requires_grad=True) for _ in range(100)]
             check(args, aot_autograd_f, f)
 
+    def test_non_tensor_args(self):
+        def fn(x, p):
+            return x * p
+
+        def fn_dropout(x, prob):
+            return torch.nn.functional.dropout(x, p=prob)
+
+        def check(a, b, mem_optimized_fn, fn):
+            a_clone = a.clone().detach().requires_grad_(True)
+            ref = fn(a, b)
+            ref.sum().backward()
+
+            res = mem_optimized_fn(a_clone, b)
```

Comment:
should probably be called something else since it's not necessary a mem_optimized_fn?